File: Binary_Search.java
import java.util.*;
class Binary_Search
{
    public void BSearch()
    {
        int a[]=new int [10];
        Scanner sc=new Scanner (System.in);
        System.out.println("Enter an array of 10 elements in Ascending order");
        for(int i=0;i<10;i++)
        a[i]=sc.nextInt();
        int n=10;
        System.out.println("Enter an element to be searched");
        int ele=sc.nextInt();
        int s=0,e=n-1,m,f=-1;
        while(s<=e)
        {
            m=(s+e)/2;
            if(a[m]==ele)
            {
                f=m;
                break;
            }
            else if(a[m]>ele)
            e=m-1;
            else if(a[m]<ele)
            s=m+1;
        }
        if(f>=0)
        {
            System.out.println("Search successful");
            System.out.println("Element :"+a[f]);
            System.out.println("Position :"+(f+1));
        }
        else
        System.out.println("Search unsuccessful");
    }
    public static void main()
    {
        Binary_Search obj=new Binary_Search();
        obj.BSearch();
    }
}

File: Binary Search.cpp
#include<stdio.h>
int search(int r,int a[],int n)
{
	int mid,lb=0,ub=r;
	while(lb<=ub)
	{
		mid=(lb+ub)/2;
		if(n==a[mid])
		{
			return mid+1;
			break;
		}
		else if(n>a[mid])
		lb=mid+1;
		else if(n<a[mid])
		ub=mid-1;
	}
	return 0;
}
int main()
{
	int n,r,mid;
	int a[50];
	printf("Enter the range of the array\n");
	scanf("%d",&r);
	printf("Enter the array elements\n");
	for(int i=0;i<r;i++)
	scanf("%d",&a[i]);
	printf("Enter the element to be searched\n");
	scanf("%d",&n);
	mid=search(r,a,n);
	if(mid)
	printf("The element is found on position: %d",mid);
	else
	printf("The element is not found");
}